### ISSUE TITLE
Sample 2 Issue

### ISSUE THREAD

**Issue by jamierytlewski** [practice-dev/practice-dev#5]
Hello! It's me again, I was trying Sample 2. Yes, again with a .NET/Azure issue. I'm assuming it has to do with CORS, but I was wondering if you could troubleshoot this:

https://practive-dev-backend.azurewebsites.net/api/PracticeDevPassword/

I have added a few domains for CORS, but I might be missing one. I added https://practice.dev and https://api.practice.dev

**Comment by tomaszzmudzinski** [comment]
I am wondering why you're not using ngrok (https://ngrok.com/) which is recommended by this platform. I solved this challenge in .NET Core 3.1 and used ngrok to make my localhost app visible on public URL. This takes less time than publishing site to azure and if your API is not working than you know it's just fault of your API, not some azure configuration 👍 😅

**Comment by jamierytlewski** [comment]
@tomaszzmudzinski - thanks for the feedback. There are a few assumptions that were made, but just to let you know I had the same issues with ngrok as well. It seemed as though maybe the server was having issues. I'm sure it'll show up in the logs as there were actual errors. 

Just to answer your questions. 

1. I'm doing a bunch of playing around, so playing with Github/Azure Actions. Your way is "takes less time", my way has more learning curves and trying to show what can be done. 
1. My work has blocked ngrok and having the ports open
1. Actually ngrok is not recommended and is also said to be slower so you might have to use `--region=us`. I read all that, but that doesn't mean there are other ways. If we want to get technical, CodeSandbox.io is recommended and ngrok is just another way they show of doing these steps. Why not have more options? 

I'm going to close this as it seemed to be an issue with the server/code at the moment as I had the same issues with ngrok, but after a few tries it worked.

**Comment by lstkz** [comment]
@jamierytlewski you don't have to configure CORS for API challenges. CORS are only needed if code is called from the browser with a different origin.

**Comment by tomaszzmudzinski** [comment]
@jamierytlewski I've checked your repository and your ValidateController is configured like this:
```
    [Route("api/[controller]")]
    [ApiController]
    public class ValidateController : ControllerBase
    {
         ...
    }
```

But when you look at API spec on this challenge you will see:

**POST /validate**

With your current configuration everything works on /api/validate. Just change route attribute to this:

`[Route("[controller]")]`

**Comment by lstkz** [comment]
@jamierytlewski Or you can provide URL `https://your-domain.com/api`

**Comment by jamierytlewski** [comment]
Thanks again. Yes I had the URL incorrect at first. Which I figured out when it started to work and I got a 404 versus some other JS errors. I know sometimes it’s hard to believe but the practice server was having issues. Like I said I was able to reproduce it when not at work. I should’ve copied the error, but between ngrok and Azure it was the same error. It was an error that when I pulled up the console window was printing an error out. 

It was while trying to make a call to api.practice.dev which I why I thought it was CORS. 

Now looking at the code it may have been a puppeteer timeout.

**Comment by lstkz** [comment]
Can you provide the link to your app (Azure link)? I will check it.

**Comment by jamierytlewski** [comment]
Don’t worry. I got it fixed. It worked last night with both ngrok and Azure. The initial problem was when the modal comes up to enter the URL it never even went to the URL, I just received an error message saying something to the effect of can’t connect. With this hitting an API I thought it might be a CORS issue. Then I tried ngrok when I got home same issue. I waited about 15 minutes and then both the ngrok and Azure service worked. So I’m assuming it was a hiccup in the server. Thanks again.  

Jamie
